src/transactions/cpfp.rs

```diff
@@ -0,0 +1,181 @@
+use crate::{
+    error::*,
+    transactions::{utils, CpfpableTransaction, RevaultTransaction, CPFP_MIN_CHANGE},
+    txins::*,
+    txouts::*,
+};
+
+use miniscript::bitcoin::{
+    consensus::encode::Decodable,
+    util::psbt::{Input as PsbtIn, PartiallySignedTransaction as Psbt},
+    Amount, Script, SigHashType, Transaction, TxIn, TxOut,
+};
+
+#[cfg(feature = "use-serde")]
+use {
+    serde::de::{self, Deserialize, Deserializer},
+    serde::ser::{Serialize, Serializer},
+};
+
+impl_revault_transaction!(
+    CpfpTransaction,
+    doc = "The transaction feebumping either an unvault or a spend"
+);
+impl CpfpTransaction {
+    /// Create a new cpfp transaction from the transaction that needs cpfp, its fees and weight;
+    /// the required feerate (expressed in sats/WU); a list of CpfpTxIn that can be used along
+    /// with their child number, the cpfp descriptor and the secp curve.
+    pub(crate) fn from_tx(
+        to_be_cpfped: &impl CpfpableTransaction,
+        feerate: f64,
+        mut listunspent: Vec<CpfpTxIn>,
+    ) -> Result<CpfpTransaction, TransactionCreationError> {
+        assert!(to_be_cpfped.is_finalized());
+        // Checking that the feerate requested is higher than the previous
+        // tx feerate (otherwise why cpfping in the first place?)
+        let to_be_cpfped_weight = to_be_cpfped.clone().into_tx().get_weight();
+        let to_be_cpfped_fees = Amount::from_sat(to_be_cpfped.fees());
```

Comment:
`fees()` should probably return an `Amount`, what do you think? (not to be done in this PR)

Reply:
Definitely :)